### DISCUSSION TITLE
Add additional query parameters to the entries page so a cpUrl function can use them

### DISCUSSION THREAD

**Discussion by Jorrit-Verstappen** [craftcms/cms#10669] [Ideas] [answered]
I am trying to create a link to the entries page with the status on 'pending' with the use of cpUrl. Sadly the entries page doesn't use additional query params as I tried giving it, 'status=pending'.

Apart from my use case I think it would be a good feature to have more customisation in the cpUrl function so additional parameters can be added that affect the active filters or status on a page for example.

**Comment by brandonkelly** [comment, marked as answer]
I’ve wanted to add this for a while, but it hasn’t been feasible due to the way element index classes replace the URL based on the selected source. However I was able to make it happen for Craft 4 thanks to some related JS refactoring.

So this has been added for the status and sort menus, for the next Craft 4 Beta release. Changing their selections will add `status` and `sort` query string params to the URL, which will be respected on initialization if present. (1f54cd2011686c10bacf3f52ca2cb3d3c1920c6b)

**Reply by brandonkelly** [reply]
Craft 4.0.0-beta.4 is out now with that change.

**Reply by Jorrit-Verstappen** [reply]
Thanks for the reply!
Not going to be available in Craft 3 then I presume, but good to hear it's coming in Craft 4
